From 9f44866b5cddc0ac9fb10b64202f3f8fefc988c3 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Wed, 8 Dec 2021 20:30:46 -0700
Subject: [PATCH] rustsec: fix the fixer ...

... the cargo-edit crate had a behavior change that now requires the
absolute path to the toml file. Otherwise cargo-audit will panic with
the message "Absolute path needed, got: "
---
 rustsec/src/fixer.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/rustsec/src/fixer.rs b/rustsec/src/fixer.rs
index 90653c7..61aeda3 100644
--- a/rustsec/src/fixer.rs
+++ b/rustsec/src/fixer.rs
@@ -15,7 +15,8 @@ pub struct Fixer {
 impl Fixer {
     /// Create a new [`Fixer`] for the given `Cargo.toml` file
     pub fn new(cargo_toml: impl AsRef<Path>) -> Result<Self, Error> {
-        let manifest = cargo_edit::LocalManifest::try_new(cargo_toml.as_ref())?;
+        let manifest =
+            cargo_edit::LocalManifest::try_new(cargo_toml.as_ref().canonicalize()?.as_ref())?;
         Ok(Self { manifest })
     }
 
-- 
2.34.1

